src/python/src/grpc/_adapter/_c/__init__.pyx

```diff
@@ -0,0 +1,992 @@
+# vim: expandtab:ts=2:sw=2:sts=2
+
+cimport grpc
+cimport cpython
+
+import time
+
+# There are a lot of nested structures in GRPC C core. We wrap all of them in
+# Cython extension classes and nest them in parallel in their C and Python
+# representations. Because they memory is synchronized between them, we can
+# forego a lot of the manual memory management that would be handled by using
+# the CPython API alone. Furthermore, we don't need to handle as many GC issues
+# (any GC issues at all?) because of this interlocking between C and Python
+# memory.
+
+#
+# Global state
```

Comment:
I'm generally against "section headings" in source files as they indicate both knowledge that a file is too large and knowledge of where to break it up. Is there a Cython reason for having all this code together?